reactfire/firestore/index.tsx

```diff
@@ -49,6 +50,23 @@ export function useFirestoreDoc<T = unknown>(
   );
 }
 
+/**
+ * Get a firestore document and don't subscribe to changes
+ *
+ * @param ref - Reference to the document you want to get
+ * @param options
+ */
+export function useFirestoreDocOnce<T = unknown>(
+  ref: firestore.DocumentReference,
+  options?: ReactFireOptions<T>
+): T extends {} ? T : firestore.DocumentSnapshot {
+  return useObservable(
+    doc(ref).pipe(first()),
+    `useFirestoreDocOnce:${ref.path}:${JSON.stringify(options)}`,
```

Comment:
Sorry #210 is a bit of a moving target. Updated the pattern again, let's switch this to `firestore:docOnce:${ref.firestore.app.name}:${ref.path}` and the other one to `firestore:docDataOnce:${ref.firestore.app.name}:${ref.path}:idField=${idField}`. That way we're ensuring we're caching the observable correctly. I'm going to write a util function that does this automatically in the future.

Reply:
Would you like me to update all the functions as above, or just the ones I have added?

Reply:
Also, presumably ${idField} may not exist?

Reply:
Yeah that won't be a problem, I just want to make sure that if the developer doesn't get unexpected results from it using a prior cached observable when they change `idField`.